tesseract_kinematics/test/kinematics_core_unit.cpp

```diff
@@ -89,6 +91,25 @@ TEST(TesseractKinematicsUnit, UtilsHarmonizeUnit)  // NOLINT
   EXPECT_NEAR(q[1], 3 * M_PI_4, 1e-6);
 }
 
+TEST(TesseractKinematicsUnit, UtilsNearSingularityUnit)  // NOLINT
+{
+  tesseract_scene_graph::SceneGraph::Ptr scene_graph = tesseract_kinematics::test_suite::getSceneGraphABB();
+
+  tesseract_kinematics::KDLFwdKinChain fwd_kin;
+  fwd_kin.init(scene_graph, "base_link", "tool0", "manip");
+
+  Eigen::VectorXd jv = Eigen::VectorXd::Zero(6);
+  Eigen::MatrixXd jacobian = fwd_kin.calcJacobian(jv);
+  EXPECT_TRUE(tesseract_kinematics::isNearSingularity(jacobian, 0.001));
+  jv[4] = 1 * M_PI / 180.0;
+  jacobian = fwd_kin.calcJacobian(jv);
+  EXPECT_TRUE(tesseract_kinematics::isNearSingularity(jacobian));
+  jv[4] = 2 * M_PI / 180.0;
+  jacobian = fwd_kin.calcJacobian(jv);
+  EXPECT_FALSE(tesseract_kinematics::isNearSingularity(jacobian));
+  EXPECT_TRUE(tesseract_kinematics::isNearSingularity(jacobian, 0.02));
+}
+
```

Comment:
I'll trust you that that is near a singularity for that robot

Reply:
Yea for the abb robot when joint 4, 5 and 6 are all zeros it is at a singularity and by changing joint 5 causes it to move away from the singularity.